test/NuGet.Core.Tests/NuGet.PackageManagement.Test/BuildIntegratedTests.cs

```diff
@@ -1848,14 +1851,2073 @@ public async Task TestPacMan_BuildIntegrated_PreviewUpdatesAsync_WithStrictVersi
+                    buildIntegratedProjects[0].ProjectReferences.Add(reference2);
+                    buildIntegratedProjects[0].ProjectReferences.Add(reference3);
+                    buildIntegratedProjects[0].ProjectReferences.Add(normalReference);
+
+                    buildIntegratedProjects[1].ProjectReferences.Add(reference2);
+                    buildIntegratedProjects[1].ProjectReferences.Add(reference3);
+                    buildIntegratedProjects[1].ProjectReferences.Add(normalReference);
+
+                    buildIntegratedProjects[2].ProjectReferences.Add(reference3);
+                    buildIntegratedProjects[2].ProjectReferences.Add(normalReference);
+
+                    // Act
+                    var results = await nuGetPackageManager.PreviewBuildIntegratedProjectsActionsAsync(
+						buildIntegratedProjects,
+                        packageA,
+                        new TestNuGetProjectContext(),
+                        sourceRepositoryProvider.GetRepositories(),
+                        CancellationToken.None);
+
+                    var actions = results.Select(a => a.Action).ToArray();
+
+                    await nuGetPackageManager.ExecuteNuGetProjectActionsAsync(
+                        buildIntegratedProjects,
+                        actions,
+                        testNuGetProjectContext,
+                        new SourceCacheContext(),
+                        CancellationToken.None);
+
+                    // Assert
+                    var builtIntegratedActions = actions.OfType<BuildIntegratedProjectAction>().ToList();
+                    
+                    Assert.Equal(actions.Count(), builtIntegratedActions.Count());
+
+                    Assert.True(builtIntegratedActions.All(b => b.RestoreResult.Success));
+
+                    Assert.True(builtIntegratedActions.All(b => !b.RestoreResult.LogMessages.Any())); // There should be no error or warning.
+
+                    // Make sure all 4 project installed packageB
+                    foreach (var buildIntegratedProject in buildIntegratedProjects)
+                    {
+                        var finalInstalledPackages = await buildIntegratedProject.GetInstalledPackagesAsync(CancellationToken.None);
+                        Assert.True(finalInstalledPackages.Any(f => f.PackageIdentity.Id == packageA.Id
+                        && f.PackageIdentity.Version == packageA.Version));
+                    }
+                    
+                    var resulting = actions.Select(a => Tuple.Create(a.PackageIdentity, a.NuGetProjectActionType)).ToArray();
+                    var expected = new List<Tuple<PackageIdentity, NuGetProjectActionType>>();
+                    Expected(expected, packageA.Id, packageA.Version);
+                    Assert.True(Compare(resulting, expected));
+                }
+            }
+            finally
+            {
+                foreach (TestDirectory projectDirectory in projectDirectories)
```

Comment:
if you use the path context directory and define the project directories out of that, then you won't need this.

Reply:
I don't see any complex projects created by pathContext directory, not sure if it work for `NetCorePackageReferenceProject` case. Please  check my ported code.